### PR TITLE
Fixed the issue of the main page getting stuck due to the long execution time required for updates

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

https://github.com/cocos/cocos-engine/issues/17544

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/extensions/assets-manager/AssetsManagerEx.h [hunk 1]
```diff
@@ -58,6 +58,7 @@ class CC_EX_DLL AssetsManagerEx : public RefCounted {
         DOWNLOADING_MANIFEST,
         MANIFEST_LOADED,
         NEED_UPDATE,
+        PARPER_UPDATING,
```
Comment (minggo) [inline native/extensions/assets-manager/AssetsManagerEx.h]: What does this event mean?

Comment (qiuguohua) [reply native/extensions/assets-manager/AssetsManagerEx.h]: Indicates that the prepare update function is being processed

Comment (minggo) [reply native/extensions/assets-manager/AssetsManagerEx.h]: Is `PARPER` a typo?

File: native/extensions/assets-manager/AssetsManagerEx.cpp [hunk 2]
```diff
@@ -648,20 +654,27 @@ void AssetsManagerEx::parseManifest() {
...
 }
 
 void AssetsManagerEx::prepareUpdate() {
-    if (_updateState != State::NEED_UPDATE) {
+    prepareUpdateAsync();
+}
+
+void AssetsManagerEx::prepareUpdateAsync(const PrepareUpdateFinishedCallback &cb) {
+    // Avoiding multiple function calls.
+    if (_updateState != State::NEED_UPDATE || _updateState == State::PARPER_UPDATING) {
```
Comment (minggo) [inline native/extensions/assets-manager/AssetsManagerEx.cpp]: It is better to print a warning log.

File: native/extensions/assets-manager/AssetsManagerEx.h (hunk 1 above)
Comment (qiuguohua) [reply native/extensions/assets-manager/AssetsManagerEx.h]: Yes, fiexed.

**Review by minggo — APPROVED** [review]

**Pull request merged**